src/core/mount.c

```diff
@@ -135,6 +135,12 @@ static bool mount_state_active(MountState state) {
                       MOUNT_REMOUNTING_SIGKILL);
 }
 
+static bool mount_is_bound_to_device(const MountParameters *p) {
+        assert(p);
+
+        return fstab_test_option(p->options, "x-systemd.device-bound\0");
+}
```

Comment:
wouldn't it be better if this call would take a "Mount *m" parameter, then look for the option in the right parameter set, and also do the check for m->from_fragment to make the thing bound then...